File: 0242. Valid Anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {

        if(s.length() != t.length()) return false;
        
        String[] strSplit1 = s.split("");
        Arrays.sort(strSplit1);

        String[] strSplit2 = t.split("");
        Arrays.sort(strSplit2);

        for(int i=0; i<s.length(); i++){
            if(!strSplit1[i].equals(strSplit2[i])) return false;
        }

        return true;
    }
}

File: 0242. Valid Anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:        
        if len(s) != len(t):
            return False

        letter_count = {}
        for c in s:
            if c in letter_count:
                letter_count[c] += 1
            else:
                letter_count[c] = 1

        for c in t:
            if c not in letter_count or letter_count[c] == 0:
                return False
            else:
                letter_count[c] -= 1
 
        return True
